src/linux/classes/com/alibaba/wisp/engine/WispTask.java

```diff
@@ -168,11 +168,18 @@ static void trackTask(WispTask task) {
     WispControlGroup controlGroup;
 
     WispTask(WispCarrier carrier, Coroutine ctx, boolean isRealTask, boolean isThreadTask) {
+        this(carrier, ctx, isRealTask, isThreadTask, WispConfiguration.STACK_SIZE);
+    }
+
+    /**
+     * If stackSize is 0 then -Dcom.alibaba.wisp.stacksize (currently 512K by default) will be used.
+     */
+    WispTask(WispCarrier carrier, Coroutine ctx, boolean isRealTask, boolean isThreadTask, long stackSize) {
         this.isThreadTask = isThreadTask;
         this.id = isRealTask ? idGenerator.addAndGet(1) : -1;
         setCarrier(carrier);
         if (isRealTask) {
-            this.ctx = ctx != null ? ctx : new CacheableCoroutine(WispConfiguration.STACK_SIZE);
+            this.ctx = ctx != null ? ctx : new CacheableCoroutine(stackSize > 0 ? stackSize : WispConfiguration.STACK_SIZE);
             this.ctx.setWispTask(id, this, carrier);
```

Comment:
There is a **coroutine caching mechanism** in wisp (see `CacheableCoroutine::run(), taskExit()`),  we should only cache it when the coroutine uses the default stack size.